http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -206,43 +206,35 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
         case None =>
           runtimeConfig.cloudService.process(runtime.id, RuntimeStatus.Deleting).compile.drain
       }
-      fa = if (msg.deleteDisk)
-        runtimeConfig match {
-          case rc: RuntimeConfig.GceWithPdConfig =>
-            for {
-              _ <- poll
-              now <- nowInstant
-              _ <- rc.persistentDiskId.traverse { id =>
-                val deleteDisk = for {
-                  diskOpt <- persistentDiskQuery.getPersistentDiskRecord(id).transaction
-                  disk <- F.fromEither(diskOpt.toRight(new RuntimeException(s"disk not found for ${id}")))
-
-                  deleteDiskOp <- googleDiskService.deleteDisk(runtime.googleProject, disk.zone, disk.name)
-                  _ <- computePollOperation.pollZoneOperation(runtime.googleProject,
-                                                              disk.zone,
-                                                              OperationName(deleteDiskOp.getName),
-                                                              2 seconds,
-                                                              10,
-                                                              None)(
-                    persistentDiskQuery.delete(id, now).transaction.void,
-                    F.raiseError(
-                      new RuntimeException(s"Fail to delete ${disk.name} in a timely manner")
-                    ),
-                    F.unit
-                  )
-                } yield ()
-
-                deleteDisk.handleErrorWith(e =>
-                  clusterErrorQuery
-                    .save(runtime.id, RuntimeError(e.getMessage, -1, now))
-                    .transaction
-                    .void
-                )
-              }
-            } yield ()
-          case _ => poll
-        }
-      else poll
+      fa = msg.persistentDiskToDelete.fold(poll) { id =>
+        val deleteDisk = for {
+          _ <- poll
+          now <- nowInstant
+          diskOpt <- persistentDiskQuery.getPersistentDiskRecord(id).transaction
+          disk <- F.fromEither(diskOpt.toRight(new RuntimeException(s"disk not found for ${id}")))
+
+          deleteDiskOp <- googleDiskService.deleteDisk(runtime.googleProject, disk.zone, disk.name)
+          _ <- computePollOperation.pollZoneOperation(runtime.googleProject,
+                                                      disk.zone,
+                                                      OperationName(deleteDiskOp.getName),
+                                                      2 seconds,
+                                                      10,
+                                                      None)(
+            persistentDiskQuery.delete(id, now).transaction.void,
```

Comment:
By the way, I updated this line in my PR.. I think the Sam disk resource needs to be deleted here as well (we currently never delete disk Sam resources). I can rebase if this goes first.

Reply:
that's good to know...I'll remember to add this if yours go in first